datanode/src/uss_metadata.py

```diff
@@ -248,13 +286,44 @@ def remove_operation(self, uss_id, gufi):
         ]
         found = (len(oper['operations']) == num_operations - 1)
         oper['version'] = self.version
-        oper['timestamp'] = self.format_ts()
+        oper['timestamp'] = format_utils.format_ts()
     return found
 
-  def format_ts(self, timestamp=None):
-    r = datetime.datetime.now(pytz.utc) if timestamp is None else timestamp
-    r = r.astimezone(pytz.utc)
-    return '{0}Z'.format(r.strftime('%Y-%m-%dT%H:%M:%S.%f')[:23])
+  def insert_uvr(self, uvr):
+    """Inserts a UVR, failing if another UVR with the same ID already exists.
+
+    Args:
+      uvr: UVR instance
+
+    Raises:
+      ValueError: if a UVR already exists with the new UVR's message_id
+    """
+
+    if uvr['message_id'] in [existing['message_id'] for existing in self.uvrs]:
+      raise ValueError('UVR with message_id %s already exists; it must be '
+                       'removed before inserting a new UVR with that ID' %
+                       (uvr['message_id']))
+
+    # Now add the new record
+    uvr['timestamp'] = format_utils.format_ts()
+    self.uvrs.append(uvr)
```

Comment:
I believe you need to update the version here. Can you also add a test to verify the version increases on insert and remove?

Reply:
Thanks for catching that.  Previous codepath always went through remove_uvr and I didn't fix the version update when I changed it to not go through remove_uvr.  I've added a test that was failing, and then I fixed the test as suggested.